custom_components/thermal_comfort/config_flow.py

```diff
@@ -0,0 +1,146 @@
+    :param default: default value for parameter, defaults to None
+    :returns: parameter value, or default value or None
+    """
+    if config_entry is not None:
+        return config_entry.options.get(param, config_entry.data.get(param, default))
+    else:
+        return default
+
+
+def build_schema(
+    config_entry: config_entries | None, show_advanced: bool = False
+) -> vol.Schema:
+    """Build configuration schema.
+
+    :param config_entry: config entry for getting current parameters on None
+    :param show_advanced: bool: should we show advanced options?
+    :return: Configuration schema with default parameters
+    """
+    # ToDo: get list of CONF_TEMPERATURE_SENSOR and CONF_HUMIDITY_SENSOR to create dropdown list and "one of" selection
+    schema = vol.Schema(
+        {
+            vol.Required(
+                CONF_NAME, default=get_value(config_entry, CONF_NAME, DEFAULT_NAME)
+            ): str,
+            vol.Required(
+                CONF_TEMPERATURE_SENSOR,
+                default=get_value(config_entry, CONF_TEMPERATURE_SENSOR),
+            ): str,
+            vol.Required(
+                CONF_HUMIDITY_SENSOR,
+                default=get_value(config_entry, CONF_HUMIDITY_SENSOR),
+            ): str,
+        }
+    )
+    if show_advanced:
+        schema = schema.extend(
+            {
+                vol.Optional(
+                    CONF_POLL, default=get_value(config_entry, CONF_POLL, False)
+                ): bool,
+            }
+        )
+        for st in SensorType:
+            default_enable = st in DEFAULT_SENSOR_TYPES
+            schema = schema.extend(
+                {
+                    vol.Optional(
+                        str(st),
+                        default=get_value(config_entry, str(st), default_enable),
+                    ): bool
+                }
+            )
+
+    return schema
+
+
+class ThermalComfortConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Configuration flow for setting up new thermal_comfort entry."""
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return ThermalComfortOptionsFlow(config_entry)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+
+        if user_input is not None:
+            er = entity_registry.async_get(self.hass)
+
+            t_sensor = er.async_get(user_input[CONF_TEMPERATURE_SENSOR])
+            p_sensor = er.async_get(user_input[CONF_HUMIDITY_SENSOR])
+            _LOGGER.debug(f"Going to use t_sensor {t_sensor}")
+            _LOGGER.debug(f"Going to use p_sensor {p_sensor}")
+
+            if t_sensor is None:
+                self.async_abort(reason="Temperature sensor not found")
+
+            if p_sensor is None:
+                self.async_abort(reason="Pressure sensor not found")
```

Comment:
Those probably aren't needed anymore if you use the Config Entry ID as fallback, but for future uses. You need to set the keyword for the parameter and return or it will not abort.